crates/pixi_manifest/src/toml/document.rs

```diff
@@ -142,7 +170,24 @@ impl TomlDocument {
             .entry(array_name)
             .or_insert(Item::Value(Value::Array(Array::new())))
             .as_array_mut()
-            .ok_or_else(|| TomlError::array_error(array_name, table_name.to_string().as_str()))
+            .ok_or_else(|| TomlError::array_error(array_name, table_name))
+    }
+
+    /// Retrieves a mutable reference to a target array `array_name`
+    /// in table specified by pre-split parts.
+    ///
+    /// If the array is not found, it is inserted into the document.
+    pub fn get_or_insert_toml_array_mut_from_parts<'a>(
+        &'a mut self,
+        table_parts: Vec<String>,
```

Comment:
I guess this can be more generic? An `impl Iterator<Item = &String>` perhaps? As you are simply joining.